File: lifecycle/src/main/java/xyz/truenight/databinding/lifecycle/BindingList.java
package xyz.truenight.databinding.lifecycle;

import java.util.ArrayList;
import java.util.Collections;

/**
 * Created by true
 * date: 21/01/2018
 * time: 15:04
 * <p>
 * Copyright © Mikhail Frolov
 */

public class BindingList extends ArrayList<ViewModelBinding> {

    public BindingList() {
    }

    public BindingList(ViewModelBinding... bindings) {
        Collections.addAll(this, bindings);
    }
}

File: lifecycle/src/main/java/xyz/truenight/databinding/lifecycle/BindingList.kt
package xyz.truenight.databinding.lifecycle

import java.util.*

/**
 * Created by true
 * date: 21/01/2018
 * time: 15:04
 *
 *
 * Copyright © Mikhail Frolov
 */

class BindingList : ArrayList<ViewModelBindingProvider<*>>, BindingProvider {

    override val items: List<ViewModelBindingProvider<*>>
        get() = this

    constructor() {}

    constructor(vararg bindings: ViewModelBindingProvider<*>) {
        Collections.addAll(this, *bindings)
    }
}
